src/Oscoin/P2P/Class.hs

```diff
@@ -7,5 +7,5 @@ import           Oscoin.P2P.Types (Msg)
 
 import           Codec.Serialise (Serialise)
 
-class Monad m => MonadBroadcast c m where
+class Monad m => MonadBroadcast c tx m where
     broadcast :: (Serialise s, Serialise tx, Hashable c tx) => Msg c tx s -> m ()
```

Comment:
I couldn't get things compiling without adding this type parameter unfortunately.

Reply:
You can fix this by adding `Serialise (PubKey c)` to the constraints of `broadcast`

Reply:
3fbe6cf